Add fallible handlers `try_filter`, `try_map`, `try_filter_map` (#39)
p0lunin/dptree: README.md

## Features

 - ✔️ Declarative handlers: `dptree::{endpoint, filter, filter_map, ...}`.

>  - ✔️ Fallible handlers (`try_filter`, `try_map`, `try_filter_map`) that short-circuit a handler chain with an error when the handler output is a `Result`.

Reviewer: ```suggestion
 - ✔️ Fallible handlers (`try_filter`, `try_map`, `try_filter_map`) that short-circuit the chain upon the handler's error.
```
